config/config.go

```diff
@@ -828,6 +836,47 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 	return nil
 }
 
+// GCESDConfig is the configuration for GCE based service discovery.
+type GCESDConfig struct {
+	// Project: The Google Cloud Project ID
+	Project string `yaml:"project"`
+
+	// Zone: The zone of the scrape targets.
+	// If you need to configure multiple zones use multiple gce_sd_configs
+	Zone string `yaml:"zone"`
+
+	// Filter: Can be used optionally to filter the instance list by other criteria.
+	// Syntax of this filter string is described here in the filter query parameter section:
+	// https://cloud.google.com/compute/docs/reference/latest/instances/list
+	Filter string `yaml:"filter,omitempty"`
```

Comment:
At the moment most of it can be done by relabelling, except filtering for instance tags.
But the next Google API change (which is currently in beta) will allow to filter for more complex data, like nested fields in their API and also labels (random key value pairs that are assigned to the instances).
Not sure how this could be done with relabelling only.